contracts/exchangeIssuance/ExchangeIssuanceLeveragedForCompound.sol

```diff
@@ -0,0 +1,1346 @@
+            _swapData
+        );
+    }
+
+    /**
+     * Acquires the required amount of collateral tokens by swapping the input tokens
+     * Does nothing if collateral and input token are indentical
+     *
+     * @param _collateralToken       Address of collateral token
+     * @param _amountRequired        Remaining amount of collateral token required to repay flashloan, after having swapped debt tokens for collateral
+     * @param _inputToken            Address of input token to swap
+     * @param _maxAmountInputToken   Maximum amount of input token to spend
+     * @param _swapData              Data (token addresses and fee levels) describing the swap path
+     *
+     * @return Amount of input token spent
+     */
+    function _swapInputForCollateralToken(
+        address _collateralToken,
+        uint256 _amountRequired,
+        address _inputToken,
+        uint256 _maxAmountInputToken,
+        DEXAdapter.SwapData memory _swapData
+    )
+    internal
+    isValidPath(
+        _swapData.path,
+        _inputToken,
+        _collateralToken
+    )
+    returns (uint256)
+    {
+        if(_collateralToken == _inputToken) return _amountRequired;
+        return addresses.swapTokensForExactTokens(
+            _amountRequired,
+            _maxAmountInputToken,
+            _swapData
+        );
+    }
+
+
+    /**
+     * Swaps the collateral tokens obtained from redemption for the selected output token
+     * If both tokens are the same, does nothing
+     *
+     * @param _collateralToken        Address of collateral token
+     * @param _collateralTokenAmount  Amount of colalteral token to swap
+     * @param _outputToken            Address of the ERC20 token to swap into
+     * @param _minAmountOutputToken   Minimum amount of output token to return to the user
+     * @param _swapData               Data (token addresses and fee levels) describing the swap path
+     *
+     * @return Amount of output token obtained
+     */
+    function _swapCollateralForOutputToken(
+        address _collateralToken,
+        uint256 _collateralTokenAmount,
+        address _outputToken,
+        uint256 _minAmountOutputToken,
+        DEXAdapter.SwapData memory _swapData
+    )
+    internal
+    isValidPath(_swapData.path, _collateralToken, _outputToken)
+    returns (uint256)
+    {
+        return addresses.swapExactTokensForTokens(
+            _collateralTokenAmount,
+            _minAmountOutputToken,
+            _swapData
+        );
+    }
+
+    /**
+     * Deposit collateral to compound to obtain collateralCToken for issuance
+     *
+     * @param _collateralToken              Address of collateral token
+     * @param _depositAmount                Amount to deposit
+     */
+    function _depositToCompound(
+        address _collateralToken,
+        uint256 _depositAmount
+    ) internal {
+        // get cTokenAddress from underlying assets
+        address cTokenAddress = _getCTokenAddess(_collateralToken);
```

Comment:
The cToken address is already in the `LeveragedTokenData` of the `DecodedParams` so you could probably just pass it through here instead of regenerating it from the underlying. This would probably be cheaper in gas as well as reduce complexity / code size.

Reply:
Seems resolved by 2bc43dfd123d9c826c22e1931048af595bb3ebc3